extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/OtTracerPropagator.java

```diff
@@ -85,7 +98,29 @@ public static OtTracerPropagator getInstance() {
     if (!spanContext.isValid()) {
       return context;
     }
-    return context.with(Span.wrap(spanContext));
+
+    Context extractedContext = context.with(Span.wrap(spanContext));
+
+    // Baggage is only extracted if there is a valid SpanContext
+    if (carrier != null) {
```

Comment:
If we are in this line, it means the extracted `SpanContext` is valid, which means the carrier is not `null`, correct?

Reply:
there is a check above:
```
if (!spanContext.isValid()) { return context; }
```
in this line SpanContext is valid, but `carrier` could be potentially `null`.